adapters/tlsio_mbedtls.c

```diff
@@ -631,12 +631,25 @@ int tlsio_mbedtls_close(CONCRETE_IO_HANDLE tls_io, ON_IO_CLOSE_COMPLETE on_io_cl
         }
         else
         {
-            tls_io_instance->tlsio_state = TLSIO_STATE_CLOSING;
-            if (tls_io_instance->tls_status == TLS_STATE_INITIALIZED)
+            if (tls_io_instance->tlsio_state == TLSIO_STATE_ERROR)
             {
-                // Tell the peer that you're going to close
-                mbedtls_ssl_close_notify(&tls_io_instance->ssl);
-                tls_io_instance->tls_status = TLS_STATE_CLOSING;
+                // forced shutdown if tls is in ERROR state
+                tls_io_instance->tlsio_state = TLSIO_STATE_CLOSING;
+                if (tls_io_instance->tls_status == TLS_STATE_INITIALIZED)
+                {
+                    mbedtls_ssl_session_reset(&tls_io_instance->ssl);
```

Comment:
should it reset the session or `close_notify()`?

Reply:
When tls is in error state : because of underling sockets not available =, we can only do dirty close  (i.e reset the tls session if initialized.)

In case when tls is in OPEN state we try to notify the peer that we are attempting to close the session

Reply:
gotcha gotcha. makes sense.